src/main/java/br/com/sbs/cubatech/category/SubCategoryApiDto.java

```diff
@@ -0,0 +1,42 @@
+package br.com.sbs.cubatech.category;
+
+import br.com.sbs.cubatech.course.CourseApiDto;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SubCategoryApiDto {
+
+    private final String name;
+    private final String urlCode;
+    private final String studyGuide;
+
+    private final List<CourseApiDto> courses;
+
+    public SubCategoryApiDto(SubCategory subCategory) {
+        this.name = subCategory.getName();
+        this.urlCode = subCategory.getUrlCode();
+        this.studyGuide = subCategory.getStudyGuide();
+        this.courses = CourseApiDto.convertAll(subCategory.getCourses());
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getUrlCode() {
+        return urlCode;
+    }
+
+    public String getStudyGuide() {
+        return studyGuide;
+    }
+
+    public List<CourseApiDto> getCourses() {
+        return courses;
+    }
+
+    public static List<SubCategoryApiDto> convertAll(List<SubCategory> subCategories){
+        return subCategories.stream().map(SubCategoryApiDto::new).collect(Collectors.toList());
```

Comment:
Aqui dá pra usar o `.toList()` do Java 16+

```suggestion
        return subCategories.stream().map(SubCategoryApiDto::new).toList();
```

Não esquecer de limpar os imports.